content-security-policy/support/testharness-helper.sub.js

```diff
@@ -20,6 +20,18 @@ function getOrigin() {
   return url.toString();
 }
 
+function getCrossOrigin() {
+  var url = new URL("http://{{domains[天気の良い日]}}:{{ports[http][0]}}/");
+  return url.toString();
+}
+
+function getSecureCrossOrigin() {
+  // Note that `http` port is used. This is to check subsumption of `self`,
+  // since ports are not wildcards and are not necessarily default ports. 
```

Comment:
Can you add a little more detail, like:

"Since `wptserve` spins up servers on non-default port, `'self'` matches `http://[host]:[specified-port]` and `https://[host]:[specified-port]`, but not `https://[host]:[https-port]`. So, we use the `http` port for this `https` origin in order to verify that a secure variant of a non-secure URL matches `'self'`".